### PR TITLE
Support EU region endpoint

### LINKED ISSUE #129: Support EU region endpoints
SendGrid added a separate API for EU servers.

https://www.twilio.com/en-us/blog/send-emails-in-eu

Support was added to the official sendgrid-python library in 6.10 via a `set_sendgrid_data_residency()` method https://github.com/sendgrid/sendgrid-python/pull/1073 (although I think this could be done in earlier versions too by setting the `host` parameter of `SendGridAPIClient`).

To support this I think we'd need a new optional setting, something like `SENDGRID_REGION`

### PR DESCRIPTION
**Pull request by mikemanger**
Adds `SENDGRID_HOST_URL` to configure the API's root URL. This fixes #129 by allowing people to change the default API endpoint.

I thought about using the `set_sendgrid_data_residency()` method added in 6.10 (https://github.com/sendgrid/sendgrid-python/pull/1073) however this works with all previous releases and should "just work" if they add more regions in the future.

One downside is that to test this you need a separate subuser and API key - I've set this as `TESTING_SENDGRID_EU_API_KEY`.

I didn't intend to include this but an edge case to the `get_payload()` not always returning string is checked before doing a string replace (this is highlighted in more recent versions of mypy).

Feedback welcome.

### CODE REVIEW

File: sendgrid_backend/mail.py [hunk 1]
```diff
@@ -212,7 +220,10 @@ def set_prop(attachment, prop_name, value):
                 filename = f"part-{uuid.uuid4().hex}{ext}"
             set_prop(sg_attch, "filename", filename)
             # todo: Read content if stream?
-            set_prop(sg_attch, "content", django_attch.get_payload().replace("\n", ""))
+            payload = django_attch.get_payload()
+            if isinstance(payload, str):
```
Comment (sklarsa) [inline sendgrid_backend/mail.py:224]: nice catch

**Comment by mikemanger** [comment]
> have you tested this locally yourself?
Yes. I've fixed the secret `TESTING_SENDGRID_EU_API_KEY` not getting passed to tox on my fork so it now passes both tests.

image

For some reason it is now showing as skipping for the original `TestPostToSendgrid.test_post()` test in this repository - did the `SENDGRID_API_KEY` secret get renamed or have I broken something?🐛

**Comment by sklarsa** [comment]
I think I set it to only run on the main/master to avoid leaking the secret in PRs

**Review by sklarsa — APPROVED** [review]
